haystack/preview/utils/filters.py

```diff
@@ -76,10 +77,15 @@ def _safe_gt(first: Any, second: Any) -> bool:
     Works only for numerical values and dates. Strings, lists, tables and tensors all raise exceptions.
```

Comment:
```suggestion
    Works only for numerical values and dates in ISO format (YYYY-MM-DD). Strings, lists, tables and tensors all raise exceptions.
```

Reply:
I think we should make a bit more evident that the date format supported is ISO only for now, so let's add this mention in the docstring of `document_matches_filters` too